### PR TITLE
Parse test reports in the GH Actions workflow

### PR DESCRIPTION
**Pull request by amorfis**
Adds "Test results" under the jobs in the workflow:

Screenshot 2023-11-22 at 18 09 14

Makes it easier to find failed tests and presents nice overview of all the tests run

### CODE REVIEW

**Comment by adamw** [comment]
Nice! How come we didn't have this before? ;)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -23,6 +23,13 @@ jobs:
       run: sbt -v compile
     - name: Test
       run: sbt -v test
+    - name: Test Report
+      uses: dorny/test-reporter@v1
+      if: success() || failure()
```
Comment (adamw) [inline .github/workflows/ci.yml:28]: isn't this a no-op? that is, can't we simply remove this?

Comment (amorfis) [reply .github/workflows/ci.yml:28]: No, it means to run this step regardless if the previous one failed or succeeded. Without it test reports wouldn'd be parsed if any test failed.

Comment (adamw) [reply .github/workflows/ci.yml:28]: Ah ok :) maybe add a comment, I think people might find it confusing (e.g. me ;) )

**Comment by adamw** [comment]
Thanks!

**Pull request merged**